src/RebaseLoopsToZero.cpp

```diff
@@ -0,0 +1,54 @@
+#include "RebaseLoopsToZero.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+
+namespace Halide {
+namespace Internal {
+
+using std::string;
+
+namespace {
+
+bool should_rebase(ForType type) {
+    switch (type) {
+    case ForType::Extern:
+    case ForType::GPUBlock:
+    case ForType::GPUThread:
```

Comment:
gpu loops absolutely need to be rebased to zero, and there's a pass inside FuseGPUThreadLoops.cpp that does it. Is that mutator now redundant?

Reply:
That mutator works a little differently. It substitutes loop_var + min, rather than makes a new let, and that seems like maybe something that might matter for another pass/logic (there's a lot of stuff happening in FuseGPUThreadLoops). I would also be careful about changing when loops get rebased to 0.

Maybe this PR should make it so `rebase_loops_to_zero` accepts a set of `ForType` that get rebased, and use that in FuseGPUThreadLoops? That would at least keep the rebasing happening at the same time, and the only change would be substitute vs. let.

Reply:
I think the new mutator is just more correct than the old one. I guess the old one happens earlier though, so we can't just rely on the new one to do the mutation. Your proposal sounds good, but I have no strong feelings one way or the other.

Reply:
I think the duplication here is minimal and there are non-minimal risks in messing with this, so I'll save it for a separate PR.